validator/tests/test_scheduler/test_schedulers_with_yaml.py

```diff
@@ -0,0 +1,472 @@
+        self._single_block_files_individually(
+            scheduler=scheduler,
+            context_manager=context_manager,
+            name='heterogeneous_workload.yaml')
+
+    def test_parallel_lifo_heterogeneous_workload(self):
+        """Tests the parallel scheduler against the
+        test_scheduler/data/heterogeneous_workload.yaml file.
+
+        Notes:
+            This yaml file has namespaced addresses with 10 batches. Some
+            batches are composed of txns with only one namespace in inputs
+            and outputs, then other batches with txns with several namespaces.
+            There are 6 failed batches.
+        """
+
+        context_manager, scheduler = self._setup_parallel_scheduler()
+        self._single_block_files_individually(
+            scheduler=scheduler,
+            context_manager=context_manager,
+            name='heterogeneous_workload.yaml',
+            lifo=True)
+
+    def test_serial_heterogeneous_workload(self):
+        """Tests the serial scheduler against the
+        test_scheduler/data/heterogeneous_workload.yaml file.
+
+        Notes:
+            This yaml file has namespaced addresses with 10 batches. Some
+            batches are composed of txns with only one namespace in inputs
+            and outputs, then other batches with txns with several namespaces.
+            There are 6 failed batches.
+        """
+
+        context_manager, scheduler = self._setup_serial_scheduler()
+        self._single_block_files_individually(
+            scheduler=scheduler,
+            context_manager=context_manager,
+            name='heterogeneous_workload.yaml')
+
+    def _single_block_files_individually(self,
+                                         scheduler,
+                                         context_manager,
+                                         name,
+                                         lifo=False):
+        """Tests scheduler(s) with yaml files that represent a single
+        block.
+
+        Notes:
+            Tests that the serial scheduler has the correct batch validity
+            and state hash, and that 1 state hash is produced.
+
+        """
+
+        file_name = self._path_to_yaml_file(name)
+        tester = SchedulerTester(file_name)
+        defined_batch_results_dict = tester.batch_results
+        batch_results, txns_to_assert_state = tester.run_scheduler(
+            scheduler=scheduler,
+            context_manager=context_manager,
+            txns_executed_fifo=not lifo)
+
+        self.assert_batch_validity(
+            defined_batch_results_dict,
+            batch_results)
+
+        for t_id in txns_to_assert_state:
+            state_and_txn_context = txns_to_assert_state.get(t_id)
+            if state_and_txn_context is not None:
+                txn_context, state_found, state_assert = state_and_txn_context
+
+                self.assertEquals(state_found, state_assert,
```

Comment:
`assertEquals` is deprecated in favor of `assertEqual` (https://docs.python.org/3.3/library/unittest.html#deprecated-aliases)

There are a few instances of this. It's not specific to this PR, but they might as well be cleaned up now.